src/test/PoolNestedActionsTest.sol

```diff
@@ -92,29 +90,26 @@ contract NestedActionExecutor is Test, PoolTestBase {
         for (uint256 i = 0; i < actions.length; i++) {
             Action action = actions[i];
             if (action == Action.NESTED_EXECUTOR_LOCK) _nestedLock();
-            else if (action == Action.SWAP_AND_SETTLE) _swap();
-            else if (action == Action.ADD_LIQ_AND_SETTLE) _addLiquidity();
-            else if (action == Action.REMOVE_LIQ_AND_SETTLE) _removeLiquidity();
-            else if (action == Action.DONATE_AND_SETTLE) _donate();
+            else if (action == Action.SWAP_AND_SETTLE) _swap(msg.sender);
+            else if (action == Action.ADD_LIQ_AND_SETTLE) _addLiquidity(msg.sender);
+            else if (action == Action.REMOVE_LIQ_AND_SETTLE) _removeLiquidity(msg.sender);
+            else if (action == Action.DONATE_AND_SETTLE) _donate(msg.sender);
             else if (action == Action.INITIALIZE) _initialize();
         }
     }
 
     function _nestedLock() internal {
-        (address locker) = manager.getLocker();
-        assertEq(locker, msg.sender);
-
+        bool locked = manager.isLockSet();
+        assertEq(locked, true);
         vm.expectRevert(abi.encodeWithSelector(IPoolManager.AlreadyLocked.selector));
         manager.lock("");
-
-        (locker) = manager.getLocker();
-        assertEq(locker, msg.sender);
+        locked = manager.isLockSet();
+        assertEq(locked, true);
     }
 
-    function _swap() internal {
-        (address locker) = manager.getLocker();
-        assertTrue(locker != address(this), "Locker wrong");
-
+    function _swap(address locker) internal {
```

Comment:
super nit but this really is now the caller right? Seems like we renamed (which I like better!)